src/adapter/graphql/src/queries/datasets/dataset.rs

```diff
@@ -87,6 +87,26 @@ impl Dataset {
         Ok(summary.kind.into())
     }
 
+    // TODO: Private Datasets: tests
```

Comment:
Note for readers: I plan to write tests a little later -- for now, I've limited myself to manual testing